[FIX] Separate out imaging-specific "common derivatives"

effigies opened the pull request:
It has been observed that the "common derivatives" has mostly meant "common across imaging modalities". This PR is an attempt to rectify this situation, in terms of not reasserting the centrality of imaging to BIDS, and in order to ensure that we're not cluttering the namespace and putting non-imaging modalities in a situation where they have to make unintuitive naming choices because the obvious names are all claimed.

With a couple of cleanups, the preprocessed data section of `05-derivatives/02-common-data-types.md` seems usable, as the primary definition is "operations that do not change the dimensionality of the data". We currently lack non-imaging examples, but I'm not sure whether it's wise to include outside of the BEP 021 effort, as any decisions that make it into a release have to be respected in future efforts.

Almost everything else is really imaging specific, so I moved it to `05-derivatives/03-imaging.md`, with minimal modification.

One thing we need to be careful about is whether an entity (<key>-<value>) definition is being constrained to have a meaning with regard to the imaging modality or constrained to have a meaning that may not apply outside imaging. I think we only introduce `space`, `desc` and `hemi` here, all of which feel (to me) reasonably extensible to other modalities where they might apply or inherently spatial.

One thing I haven't touched is **Coordinate systems** in `05-derivatives/01-introduction.md`. It seems applicable outside imaging, so it feels premature to move it into an imaging-specific section. Some parts might be sensibly moved here, though. Please comment here for portions that seem best moved, and in #265 for portions that should simply be reworded.

If we can keep this PR focused on conceptual clarity, we can reserve polish and typos for when #265 is nearly ready to merge.

@bids-standard/derivatives I would appreciate all of your thoughts. In particular @bids-standard/derivatives-electrophys I would like your input to ensure that this is moving in a direction that works for you.

effigies commented on src/05-derivatives/02-common-data-types.md:
```diff
@@ -59,253 +58,5 @@ pipeline1/
-static volume, a SamplingFrequency property would no longer be relevant). In
-addition, all processed files include the following metadata JSON fields:
+static volume, a SamplingFrequency property would no longer be relevant).
 
```
```suggestion
```

effigies commented on src/05-derivatives/03-imaging.md:
```diff
@@ -0,0 +1,268 @@
+| ------------- | ----------------------------------------------------------------------------------------------- |
+| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not. |
+
+
```
```suggestion
```

effigies commented on src/05-derivatives/03-imaging.md:
````diff
@@ -0,0 +1,268 @@
+102     "Brainstem"     BS      #36de72	    11
+```
+
+
````
```suggestion
```

effigies commented on src/05-derivatives/03-imaging.md:
```diff
@@ -0,0 +1,266 @@
+# Imaging data types
```
What does imaging mean to you? The sense in which I'm using it is "at least one dimension (and usually two or three) have a spatial interpretation". I don't know that these things apply to N-D data generally.

Can you clarify your objections?

robertoostenveld replied:
The term "N-Dimensional digital data" also applies to raw EEG (space and time), to surround audio (space and time), to photo (2D space), video (2D pixel space and time), eye tracker (which eye, pupil versus gaze, and time), polhemus electrode positions (space), motion capture (which body part, and time), touch screen events in behavioral data (space and time). The subsequent text does not provide  guidelines how to deal with those data types. It does not even specify how the temporal dimension of MRI is dealt with, only deals with the different spatial representations of MRI data (volumetric, surface-based, parcellated). 

Surface-based and parcellated descriptions of data are by themselves not images any more, but are derived from imaging data. That is why I consider "Imaging data types"  an appropriate title. Or perhaps "Data types derived from images". 

Interestingly, "Data types derived from images" would also apply (in case BEP021 moves further ahead) to volume conduction models of the head and source models, used in EEG and MEG pipelines, and the functional estimates that are obtained for those source models (similar to how we store HCP MEG results in CIFTI). These results are multi-modal, since the input is anatomical MRI and electrophys.

oesteban replied:
@effigies sorry for the lack in clarity. What I mean is that traversing over the whole taxonomy of the dot product of data types and modalities seems impractical.

I agree common-derivatives needs to be extra-careful not to impose some particular modality idiosyncrasy to the rest of them. To address that, the original BIDS spec only splits by modalities in a very broad sense and then BEPs followed that approach. I just can't see why this extra granularity helps to get common-derivatives merged faster or helps to better describe derivatives. IMHO it will be much harder to keep the consistency across BEPs, and also, this increases communication and review costs at no clear benefit.

robertoostenveld replied:
The derivatives should also allow the representation and description of processed `physio` data, which can be (on its own) in the `beh` folder according to the current spec. Physio can for example be GSR, ECG or optical heart signals, in relation to events. So rather than `N>1` it should be `N=>1`. 

And how about processed behavioral data represented in the (non-continuous) events? Is it appropriate to describe irregularly sampled data as N-D?

oesteban replied:
> The derivatives should also allow the representation and description of processed `physio` data, which can be (on its own) in the `beh` folder according to the current spec. Physio can for example be GSR, ECG or optical heart signals, in relation to events. So rather than `N>1` it should be `N=>1`.
> 
> And how about processed behavioral data represented in the (non-continuous) events? Is it appropriate to describe irregularly sampled data as N-D?

Sorry, I can't understand what you are suggesting here. You would not agree with _Imaging derivates_ as the main header for this split?

robertoostenveld commented on src/05-derivatives/02-common-data-types.md:
```diff
@@ -1,6 +1,6 @@
 
-## Processed, coregistered and/or resampled volumes
+## Preprocessed or cleaned data
 
```
I believe the template not only applies to `anat|func|dwi`, but also to `meg|eeg|ieeg`. This is included further down in the example as time-domain filtering.

effigies replied:
Should we give up the `anat|func|dwi|meg|...` here and go with `<datatype>/`? (I think that's the word we use for this level.) This really should apply to all.

oesteban replied:
I think `<datatype>/` (or any similar language) would be best (thinking of `connectivity/` from BEP017)

robertoostenveld commented on src/05-derivatives/02-common-data-types.md:
```diff
@@ -11,24 +11,23 @@ Template:
+ -  Motion-corrected, temporally denoised, and transformed to MNI space BOLD series
+ -  Inhomogeneity corrected and skull stripped T1w files
+ -  Motion-corrected DWI files
  -  Time-domain filtered and ICA cleaned EEG data
```
Please split "Time-domain filtering for EEG data" from "Spatial filtering, e.g. ICA cleaned EEG data" in the example, since these deal with different dimensions.

robertoostenveld replied:
the application of Maxfilter on MEG data could also be considered an example of preprocessing in which the dimensions don't change. Maxfilter is mentioned in the main spec [here](https://bids-specification.readthedocs.io/en/stable/99-appendices/06-meg-file-formats.html#neuromagelektamegin) and specifies the `proc` keyword (e.g. `proc-sss`) to be used in the derivatives.

effigies replied:
`proc-sss`... This is unfortunate. I don't really know what to do with derivatives specified prior to the derivatives spec.

robertoostenveld commented on src/05-derivatives/02-common-data-types.md:
```diff
@@ -11,24 +11,23 @@ Template:
+coordinate systems or registered to different reference maps.
+The `desc` keyword is a general purpose field with freeform values.
+To distinguish between multiple different versions of processing for the same
+input data the `desc` keyword should be used.
```
The phrasing here is not specific enough. It states that 'space` is recommended (I find that OK), and it states that `desc` simply exists (i.e. not recommended, required or whatever).

>To distinguish between ... keyword should be used.

Is that "should" a "SHOULD" according to [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)? 

Is it required/recommended/suggested (i.e. MUST/SHOULD/MAY) to use either one of `desc` or `space`? What about the existing specification of `proc`?

robertoostenveld replied:
I think that this comment can be ignored in relation to the specific PR; I guess it is better discussed in the top-level  common-derivatives PR.

sappelhoff replied:
I would add a brief part on what `desc` is an abbreviation for (=description?). This helps the reader with grasping and remembering the entity

robertoostenveld commented on src/05-derivatives/02-common-data-types.md:
```diff
@@ -11,24 +11,23 @@ Template:
-Note that even though `space` and `desc` are
-optional at least one of them needs to be defined to avoid name conflict with
-the raw file.
+Note that even though `space` and `desc` are optional at least one of them needs
```
the rigidity of "needs" is not clear wrt RFC2119.

edickie commented on src/05-derivatives/03-imaging.md:
````diff
@@ -0,0 +1,266 @@
+            sub-001_dparc.dlabel.nii
+```
+
+### Anatomical Labels
````
Yeah - I think there was some confusion between whether labels (i.e. atlases in masks) were relevant to info for the functional and dwi pipelines and therefore needed to be considered common principles. (Although I 

But I think your right. It makes more sense to logically separate Anatomical and everything that specifically does in an "anat" derivatives folder into the structural derivatives BEP.

One thing that I don't know how clear we are about if how link this info to derived files that related to these masks (i.e. parcellated functional data)

effigies replied:
I'm happy to move structural out... I think during the first big merge, the structural stuff was considered so integral to working with functional and diffusion (e.g., resampling ROIs into the bold/dwi images' spaces) that they became effectively "modality agnostic". But sure. Does it make sense to make that a separate PR or just reorganize this one a bit?

oesteban replied:
Separate PR :+1:

effigies commented on src/05-derivatives/03-imaging.md:
```diff
@@ -0,0 +1,266 @@
+
+| **Key name**  | **Description**                                                                                 |
+| ------------- | ----------------------------------------------------------------------------------------------- |
+| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not. |
```
Can this be part of the common derivatives discussion? I don't know how many people are paying attention to this PR which is just a restructuring. This decision was made by the main body of people at the workshop, so I'm hesitant to quickly discard it.

oesteban replied:
Sure, no problem.

effigies commented on src/05-derivatives/02-common-data-types.md:
````diff
@@ -11,24 +11,23 @@ Template:
 ```
 
-Processing in this context means transformations of data that does not change
+Preprocessing in this context means transformations of data that do not change
````
It's not a "preprocessed BOLD file", correct. It's still a derivative, but needs to be treated separately from a preprocessed `bold`, `dwi`, `T1w`, `eeg`, etc. file.

oesteban replied:
I still don't understand the rationale, but I guess I should raise the comment after this PR is merged, as this is just a re-organization.

effigies commented:
@robertoostenveld I think I've addressed your comments, if you'd like to have another look.

@oesteban Is there a practical suggestion you're making, wrt https://github.com/bids-standard/bids-specification/pull/310#discussion_r319313853? I'm not seeing how to operationalize it. Is it some text you're looking for or a reorganization?

@edickie @oesteban Regarding the anatomical stuff, we have masks, dsegs and probsegs in here. While segmentations/parcellations are anatomically based, the idea here was that you might want them in other spaces, which made them "common". We can pull them out, which would leave us with masks. Let me know how you want to proceed.

effigies commented:
Also, the linkchecker failure is a result of https://loni.usc.edu/ letting their cert expire.

effigies commented:
@robertoostenveld @oesteban @edickie Just a bump on this. Sorry to bug you, but I want to try to keep this moving. If you're busy, can you give me a rough time before I should check back in?

oesteban commented on src/05-derivatives/03-imaging.md:
```diff
@@ -0,0 +1,266 @@
+
+## Preprocessed, coregistered and/or resampled volumes
+
+Volumetric preprocessing does not modify the number of dimensions, and so
```
For the common-derivatives branch. Please dismiss this comment.
